packages/metrics/src/l2/l2Metrics.service.ts

```diff
@@ -0,0 +1,62 @@
+import { isNativeError } from "util/types";
+import { BaseError } from "viem";
+
+import { ZKChainProvider } from "@zkchainhub/chain-providers";
+import { ILogger } from "@zkchainhub/shared";
+
+/**
+ * Acts as a wrapper around Viem library to provide methods to interact with zkSync chains.
+ */
+export class L2MetricsService {
+    constructor(
+        private readonly provider: ZKChainProvider,
+        private readonly logger: ILogger,
+    ) {}
+
+    /**
+     * Retrieves the transactions per second (TPS) from the provider.
+     *
+     * @returns A promise that resolves to the number of transactions per second.
+     */
+    async tps(): Promise<number> {
+        return this.provider.tps();
+    }
+
+    /**
+     * Retrieves the average block time from the provider.
+     *
+     * @returns A promise that resolves to the average block time as a number.
+     */
+    async avgBlockTime(): Promise<number> {
+        return this.provider.avgBlockTime();
+    }
+
+    /**
+     * Retrieves the number of the last block in the chain.
+     *
+     * @returns A promise that resolves to a bigint representing the number of the last block.
+     */
+    async lastBlock(): Promise<bigint> {
+        return this.provider.getBlockNumber();
+    }
+
+    /**
+     * Retrieves the last verified block based on the given lastVerifiedBatch.
+     *
+     * @param lastVerifiedBatch The number representing the last verified batch.
+     * @returns A Promise that resolves to the number of the last verified block, or undefined if an error occurs.
+     */
+    async getLastVerifiedBlock(lastVerifiedBatch: number): Promise<number | undefined> {
+        try {
+            const [, endBlock] = await this.provider.getL1BatchBlockRange(lastVerifiedBatch);
+            return endBlock;
+        } catch (error) {
+            if (error instanceof BaseError) {
+                this.logger.error(error.message);
+            } else if (isNativeError(error)) {
+                this.logger.error(error);
+            }
+            return undefined;
```

Comment:
Following the way we are handling errors, for the rest of the methods. Should we throw here instead of returning undefined ? The lastVerifiedBlock is not an optional method on the front end

Reply:
this got me thinking tbh, should we actually return return value | undefined for all methods? like an error on L2 shouldn't break anything as it's like not having rpc urls wdyt?